src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -343,9 +345,25 @@ public ConfidentialClientApplicationBuilder WithTelemetryClient(params ITelemetr
                 Config.TelemetryClients = telemetryClients;
             }
 
+            TelemetryClientLogMsalVersion();
+
             return this;
         }
 
+        private void TelemetryClientLogMsalVersion()
+        {
+            MsalTelemetryEventDetails telemetryEventDetails = new MsalTelemetryEventDetails(TelemetryConstants.ConfigurationUpdateEventName);
+            telemetryEventDetails.SetProperty(TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion());

```

Comment:
The way SAL does this, which I prefer, is to not construct this until you know there's a client who would receive it. i.e. to put it inside the foreach/if blocks and only create the event details there.

Here it's pretty trivial extra work (but it is extra work) but it's a good pattern to set when emitting telemetry for the library.